components/event-signup.js

```diff
@@ -0,0 +1,213 @@
+"use client";
+
+import { useForm } from "react-hook-form";
+import { useState, useEffect } from "react";
+
+function EventSignup(eventProps) {
+  const [eventData, setEventData] = useState();
+  const [registerSuccess, setRegisterSuccess] = useState();
+  const [ticketUrl, setTicketUrl] = useState("");
+
+  useEffect(() => {
+    checkSlug();
+  }, []);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitSuccessful, isSubmitting },
+  } = useForm({
+    mode: "onTouched",
+  });
+
+  const checkSlug = async () => {
```

Comment:
You made this async but you are using promises. Either is fine to use async/await or promises, doesn't really matter to me honestly.

Reply:
Resolved in commit: f8d00b290190a9503a791f3507c28f0101cecc88